oshi-core/src/main/java/oshi/hardware/common/AbstractUser.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Oshi (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2018 The Oshi Project Team
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ *
+ * Contributors:
+ * https://github.com/oshi/oshi/graphs/contributors
+ */
+package oshi.hardware.common;
+
+import oshi.hardware.User;
+
+/**
+ * An abstract user class having information about the current user.
+ *
+ * @author : BilalAM
+ */
+public abstract class AbstractUser implements User {
+        private String userName;
+        private int userId;
+        private String homeDir;
+
+        public AbstractUser(String userName, int userId, String homeDir, String realName) {
```

Comment:
I know you're copying other existing constructors in OSH but I'm really trying to lean towards new versions not having constructors with arguments like this. Just construct the object with default values for the class variables, and have setters for them (you can keep the setters `protected` to limit API abilities.)